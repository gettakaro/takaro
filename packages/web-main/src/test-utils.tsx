import { ThemeProvider } from 'styled-components';
import { GlobalStyle, SnackbarProvider, darkTheme } from '@takaro/lib-components';

import { render, RenderOptions } from '@testing-library/react';
import { QueryClientProvider } from '@tanstack/react-query';
import { queryClient } from 'App';
import { FC, PropsWithChildren } from 'react';
import { MemoryRouter } from 'react-router-dom';

window.scrollTo = () => {};

const providers: FC<PropsWithChildren> = ({ children }) => {
  // add portal for drawer
  const drawerPortalRoot = document.createElement('div');
  drawerPortalRoot.setAttribute('id', 'drawer');
  document.body.appendChild(drawerPortalRoot);

  // add portal for dialog
  const dialogPortalRoot = document.createElement('div');
  dialogPortalRoot.setAttribute('id', 'dialog');
  document.body.appendChild(dialogPortalRoot);

  // replace getComputedStyle with a mock that returns the values we need
  const { getComputedStyle } = window;
  window.getComputedStyle = (elt) => getComputedStyle(elt);

  return (
    <ThemeProvider theme={darkTheme}>
      <QueryClientProvider client={queryClient}>
        <MemoryRouter>
          <SnackbarProvider>
            <GlobalStyle />
            {children}
          </SnackbarProvider>
        </MemoryRouter>
      </QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: providers, ...options });

export * from '@testing-library/react';
export { customRender as render };
