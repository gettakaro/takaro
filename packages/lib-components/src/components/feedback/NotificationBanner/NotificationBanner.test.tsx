import { NotificationBanner } from '.';
import { render } from 'test-utils';

it('Should render <NotificationBanner/>', () => {
  // we need to create the portal
  const notificationBannerPortal = document.createElement('div');
  notificationBannerPortal.setAttribute('id', 'notification-banner');
  document.body.appendChild(notificationBannerPortal);

  const { container } = render(
    <NotificationBanner description="Notification banner description" title="Notification banner title" />
  );
  expect(container).toMatchSnapshot();
});
