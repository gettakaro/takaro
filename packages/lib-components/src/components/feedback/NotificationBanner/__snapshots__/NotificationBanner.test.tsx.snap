// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render <NotificationBanner/> 1`] = `<div />`;
